### PR TITLE
Update migration guide for remote certificate validation check for DPS

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: SDK v2 migration guide.md [hunk 1]
```diff
@@ -428,7 +428,7 @@ These span across all service clients.
  which can be set via `ProvisioningClientOptions.RetryPolicy`.
  - ProvisioningRegistrationSubstatus.ReprovisionedToInitalAssignment value added meaning the device has been reprovisioned to a previously assigned IoT hub.
  - `DeviceRegistrationResult` now has a method `TryGetPayload<T>(...)`.
- - `ProvisioningClientTransportSettings` and derived classes now have settable property `CertificateRevocationCheck`.
+ - `ProvisioningClientTransportSettings` and derived classes now have settable property `CertificateRevocationCheck`. For connections over AMQP, this check is used in conjunction with the property `RemoteCertificateValidationCallback` on `ProvisioningClientTransportSettings`. For connections over MQTT, this check is set directly on the TLS settings.
```
Comment (drwill-ms) [inline SDK v2 migration guide.md]: Consider using sub-bullets to make it a small block of text.

```suggestion
 - `ProvisioningClientTransportSettings` and derived classes now have settable property `CertificateRevocationCheck`.
   - For connections over AMQP, this check is used in conjunction with the property `RemoteCertificateValidationCallback` on `ProvisioningClientTransportSettings`.
     - For connections over MQTT, this check is set directly on the TLS settings.
```

File: SDK v2 migration guide.md [hunk 2]
```diff
@@ -428,7 +428,9 @@ These span across all service clients.
  which can be set via `ProvisioningClientOptions.RetryPolicy`.
  - ProvisioningRegistrationSubstatus.ReprovisionedToInitalAssignment value added meaning the device has been reprovisioned to a previously assigned IoT hub.
  - `DeviceRegistrationResult` now has a method `TryGetPayload<T>(...)`.
- - `ProvisioningClientTransportSettings` and derived classes now have settable property `CertificateRevocationCheck`.
+ - `ProvisioningClientTransportSettings` and derived classes now have settable property `CertificateRevocationCheck`. Remote certificate validation check is turned off by default.
+   -  For connections over AMQP, this check is used in conjunction with the property `RemoteCertificateValidationCallback` on `ProvisioningClientTransportSettings`.
+   -   For connections over MQTT, this check is set directly on the TLS settings.
```
Comment (abhipsaMisra) [inline SDK v2 migration guide.md:433]:
```suggestion
   -  For connections over AMQP, this check is used in conjunction with the property `RemoteCertificateValidationCallback` on `ProvisioningClientTransportSettings`.
   -  For connections over MQTT, this check is set directly on the TLS settings.
```

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**